### PR TITLE
Added PWA support spec

### PR DESCRIPTION
**Pull request by amrutha95**
Added PWA support spec for community review

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/winget-cli/pull/563)

### CODE REVIEW

File: doc/specs/#194 - PWA Support.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+## Inspiration
+
+PWA adoption is extremely important in the Windows app ecosystem. It is also crucial that they are treated as first class citizens in the app ecosystem, just like any native application. Providing PWA support for winget is key to achieving this goal.
+
+## Solution Design
+
+### System requirements
+
+Windows 10 v2004 or newer
```
Comment (denelon) [inline doc/specs/#194 - PWA Support.md:24]: The client is intended to work back to 1709, how will this be validated in the client to ensure users on earlier unsupported versions of Windows 10?

File: doc/specs/#194 - PWA Support.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+Windows 10 v2004 or newer  
+Edge latest Canary build
+
+### PWA YAML manifest
+
+The "InstallerType" field in the YAML manifest file specifies the type of the application. A new InstallerType "PWA" will be added to show that a given application is a PWA.
+
+Each PWA would have its own manifest YAML file. A sample manifest file would look like this:
+
+```Id: PWATest.TestInstaller
```
Comment (denelon) [inline doc/specs/#194 - PWA Support.md:33]: CRLF missing after '''

File: doc/specs/#194 - PWA Support.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Version: 1.0.0.0
+Name: AppInstaller FinTimes
+Publisher: Microsoft Corporation
```
Comment (denelon) [inline doc/specs/#194 - PWA Support.md:36]: The `Id` should match the publisher. In this case should the `Id` be "Microsoft.FinTimes"?

File: doc/specs/#194 - PWA Support.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+There are many more optional parameters that can be passed to this service. For more information, refer to the test website linked in the Resources section.
+
+### Package installation
+
+The unsigned MSIX package and any other dependency packages are added for the current user and the application is silently installed on the user's system. The installation is done by calling the Windows API AddPackageByUriAsync() using the specific deployment options that allow unsigned packages.
+
+### Flow of the install process
+
+1. User types the install command, eg. `winget install fintimes` (For the experimental version, the manifest will be passed as an argument)
+2. The URL from the manifest YAML file of the Financial Times PWA will be sent as a POST request to an internal web service that will generate an unsigned MSIX package. THis MSIX package is downloaded onto the user's system.
```
Comment (denelon) [inline doc/specs/#194 - PWA Support.md:67]: Typo "THis" -> "This"

File: doc/specs/#194 - PWA Support.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+There are many more optional parameters that can be passed to this service. For more information, refer to the test website linked in the Resources section.
+
+### Package installation
+
+The unsigned MSIX package and any other dependency packages are added for the current user and the application is silently installed on the user's system. The installation is done by calling the Windows API AddPackageByUriAsync() using the specific deployment options that allow unsigned packages.
+
+### Flow of the install process
+
+1. User types the install command, eg. `winget install fintimes` (For the experimental version, the manifest will be passed as an argument)
```
Comment (denelon) [inline doc/specs/#194 - PWA Support.md:66]: In order to have this command work, either the Name or AppMoniker needs to be an exact match.

File: doc/specs/#194 - PWA Support.md (hunk 1 above)
Comment (JohnMcPMS) [reply doc/specs/#194 - PWA Support.md:24]: If the YAML manifest indicates the proper minimum OS version, the client will handle the check and enforcement.  And no, it is not in the index at this time.

File: doc/specs/#194 - PWA Support.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+This should have no direct impact on accessibility.
+
+### Security
+
+No direct impact on security.
+
+### Reliability
+
+This is not expected to impact reliability. It will be available as an experimental feature so users can disable it any time.
```
Comment (JohnMcPMS) [inline doc/specs/#194 - PWA Support.md:101]: The spec is not for the experimental feature, it should be for the released one.  The experimental stage is along the path.

File: doc/specs/#194 - PWA Support.md (hunk 1 above)
Comment (amrutha95) [reply doc/specs/#194 - PWA Support.md:24]: This implementation for now supports all VB+ devices. There is some ongoing work on PWA installs on VB- as well, which can be integrated in the future.

File: doc/specs/#194 - PWA Support.md (hunk 6 above)
Comment (amrutha95) [reply doc/specs/#194 - PWA Support.md]: Okay, I removed that sentence on `[this](https://github.com/microsoft/winget-cli/pull/563/commits/5b642f55005a3d5bcede69322e08a654f733ab96)`  commit. Had to rename the file there was a mismatch in the issue number.

File: doc/specs/#164 - PWA Support.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+Publisher: Microsoft Corporation
+AppMoniker: fintimes
+License: Test
+InstallerType: PWA
+MinOSVersion: 10.0.0.0
+Installers:
+- Arch: x64
+  Url: https://app.ft.com
+  InstallerType: PWA
+  ManifestVersion: 0.1.0
```
Comment (yao-msft) [inline doc/specs/#164 - PWA Support.md:46]: Nit: the ManifestVersion is a root level field

Comment (amrutha95) [reply doc/specs/#164 - PWA Support.md]: Good catch! Thank you

**Review by JohnMcPMS — APPROVED** [review]

**Review by Suirtimed — APPROVED** [review]

**Pull request merged**